src/fields/field_fe.cc

```diff
@@ -223,6 +284,59 @@ void FieldFE<spacedim, Value>::value_list (const Armor::array &point_list, const
 
 
 
+template <int spacedim, class Value>
+void FieldFE<spacedim, Value>::cache_update(FieldValueCache<typename Value::element_type> &data_cache,
+		ElementCacheMap &cache_map, unsigned int region_idx)
+{
+    ASSERT( !boundary_dofs_ ).error("boundary field NOT supported!!\n");
+    std::shared_ptr<EvalPoints> eval_points = cache_map.eval_points();
+    Armor::ArmaMat<typename Value::element_type, Value::NRows_, Value::NCols_> mat_value;
+
+    if (fe_values_[0].dim() == -1) {
+        // initialize FEValues objects (when first using)
+        this->init_fe_value<1>(eval_points);
+        this->init_fe_value<2>(eval_points);
+        this->init_fe_value<3>(eval_points);
+    }
+
+    auto update_cache_data = cache_map.update_cache_data();
+    std::unordered_map<unsigned int, typename ElementCacheMap::RegionData>::iterator reg_elm_it =
+            update_cache_data.region_cache_indices_map_.find(region_idx);
+
+    for (unsigned int i_elm=0; i_elm<reg_elm_it->second.element_set_.size(); ++i_elm) {
```

Comment:
Over-complicated ElementCacheMap.
Why there is ElementSet in the RegionData? I think there is no need for RegionData at all. As cached elements of the single region form a continuous sequence (range) then 
`elm_idx_`  + `UpdateCacheHelper::region_cache_indices_range_`

contains whole information.

Would be better to pass the region_chunk index (`UpdateCacheHelper::region_cache_indices_range_`)
 instead of the region_idx in order to avoid unecessary `find(region_idx)`. 
Then this loop should probably run over the region chunk using `region_cache_indices_range_`, similarly to FieldModel<..>::cache_update

Reply:
Mark as an auxiliary internal structure. We can probably replace it by a map:
region_index -> region_cache_range_index. Postpone further optimization after the FeildFE will be actually used.